compiler/src/iree/compiler/Dialect/Flow/Transforms/ConvertConv2D1x1ToMatmul.cpp

```diff
@@ -20,21 +20,51 @@ namespace Flow {
 namespace {
 
 // Converts linalg.conv_2d_input_nhwc_filter_nhwc op to linalg.matmul
-class Convert1x1ConvolutionMatmulOp
-    : public OpRewritePattern<linalg::Conv2DNhwcHwcfOp> {
+template <typename Conv2DOpType>
+class Convert1x1ConvolutionMatmulOp : public OpRewritePattern<Conv2DOpType> {
  public:
-  using OpRewritePattern<linalg::Conv2DNhwcHwcfOp>::OpRewritePattern;
+  using OpRewritePattern<Conv2DOpType>::OpRewritePattern;
 
-  LogicalResult matchAndRewrite(linalg::Conv2DNhwcHwcfOp convOp,
+  LogicalResult matchAndRewrite(Conv2DOpType convOp,
                                 PatternRewriter &rewriter) const override {
     RankedTensorType inputShapeType =
-        convOp.getInputOperand(0)->get().getType().dyn_cast<RankedTensorType>();
+        convOp.getInputOperand(0)
+            ->get()
+            .getType()
+            .template dyn_cast<RankedTensorType>();
     RankedTensorType filterShapeType =
-        convOp.getInputOperand(1)->get().getType().dyn_cast<RankedTensorType>();
-    RankedTensorType outputShapeType = convOp.getOutputOperand(0)
-                                           ->get()
-                                           .getType()
-                                           .dyn_cast<RankedTensorType>();
+        convOp.getInputOperand(1)
+            ->get()
+            .getType()
+            .template dyn_cast<RankedTensorType>();
+    RankedTensorType outputShapeType =
+        convOp.getOutputOperand(0)
+            ->get()
+            .getType()
+            .template dyn_cast<RankedTensorType>();
+
+    const bool isNCHW = isa<linalg::Conv2DNchwFchwOp>(convOp);
+    const bool isNHWC = isa<linalg::Conv2DNhwcHwcfOp>(convOp);
+    if (!isNCHW & !isNHWC) return failure();
+
+    // Adjusting dimension indices based on Conv2DOpType.
+    const int kcIndex = isNHWC ? 2 : 1;
+    const int kfIndex = isNHWC ? 3 : 0;
+    const int khIndex = isNHWC ? 0 : 2;
+    const int kwIndex = isNHWC ? 1 : 3;
+    const int ohIndex = isNHWC ? 1 : 2;
+    const int owIndex = isNHWC ? 2 : 3;
+    const int ocIndex = isNHWC ? 3 : 1;
+
+    SmallVector<ReassociationIndices, 4> reassociationInputOutputIndices;
```

Comment:
Move these to be after all checks that `return failure()`? (So that we don't need to build these vectors if the pattern won't apply at all.)

Reply:
done